pkg/registry/core/pod/strategy.go

```diff
@@ -334,6 +342,7 @@ func (podResizeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.
 	*newPod = *dropNonResizeUpdates(newPod, oldPod)
```

Comment:
dropNonResizeUpdates → dropPodUpdates looks like it will erase managedField updates, which doesn't seem quite right

also (pre-existing) dropPodUpdates looks really weird... it *only* preserves name / namespace / resourceVersion / uid from the new pod, but 3 of those (name / namespace / uid) are immutable fields ... why not just start with the entire old pod as a starting point?

Reply:
Can we address this in a separate PR?

Reply:
yup, just wanted to flag it before I forgot, separate issue looks fine and looks non-intersecting with these changes